llvm/lib/Transforms/Instrumentation/SYCLITTAnnotations.cpp

```diff
@@ -0,0 +1,295 @@
+//===---- SYCLITTAnnotations.cpp - SYCL Instrumental Annotations Pass -----===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// A transformation pass which adds instrumental calls to annotate SYCL
+// synchronization instructions. This can be used for kernel profiling.
+//===----------------------------------------------------------------------===//
+
+#include "llvm/Transforms/Instrumentation/SYCLITTAnnotations.h"
+
+#include "llvm/IR/Function.h"
+#include "llvm/IR/InstIterator.h"
+#include "llvm/IR/Instruction.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/Type.h"
+#include "llvm/InitializePasses.h"
+
+/** Following instrumentations will be linked from libdevice:
```

Comment:
```suggestion
/**
```
Let's avoid hardcoding specific library names. 
Feel free to add this to the design document instead.